internal/logprocessor/logprocessor.go

```diff
@@ -61,8 +62,14 @@ func (c *Client) parseResponse(body []byte) ([]byte, error) {
 
 	conn := connection{}
 
-	conn.SrcIP = rb.XFwdFor
-	validIP := net.ParseIP(rb.XFwdFor)
+	candidate := rb.XFwdFor
+	if strings.Contains(rb.XFwdFor, ",") {
+		parsedIPs := strings.Split(rb.XFwdFor, ",")
+		candidate = parsedIPs[0]
+	}
+
+	conn.SrcIP = candidate
```

Comment:
I think this is valid, but I would say that I dont think you need to set (possibly) conn.SrcIP twice. Here, no matter if candidate is valid or not you set conn.SrcIp to candidate, then you check if candidate is valid and if not change it to rb.SrcIP. If you reworked your logic to set conn.SrcIP to rb.SrcIP first then only update it if candidate is valid then your would only update (possibly)  conn.SrcIP once.

However I dont think that is a blocker. If you want to button it up go for it otherwise it looks ok, unless I am missing something.

Reply:
You have a possibility of setting twice either way. With also having a possibility of only setting it once either way. This way we have done as much as we can to make sure that the candidate is a valid ip before we validate it, so it should have to switch less often. Assuming that X-Forwarded-For is sent at least most of the time.